File: fsharp/grains/Grains.fs
module Grains

let square (n: int): Result<uint64,string> = 
    match n with
    | _ when n <= 0 || n > 64 -> Error "Invalid input"
    | _ -> pown 2UL (n - 1) |> Ok

let total: Result<uint64,string> = [1..64] |> Seq.sumBy (fun x -> pown 2UL (x - 1)) |> Ok

File: go/grains/grains.go
package grains

import (
	"errors"
)

// Square returns the number of grains on a given square
func Square(index int) (uint64, error) {
	if index < 1 || index > 64 {
		return 0, errors.New("Invalid index, must be between 1 and 64")
	}
	power := uint64(index - 1)
	return 1 << power, nil
}

// Total returns the total number of grains on an entire board
func Total() uint64 {
	var sum uint64
	for i := 1; i <= 64; i++ {
		count, error := Square(i)
		if error != nil {
			panic(error)
		}
		sum += count
	}
	return sum
}
